File: leetcode-questions/java/src/test/java/com/leetcode/questions/longestsubstringwithoutrepeatingcharacters/LongestSubstringWithoutRepeatingCharactersTest.java
package com.leetcode.questions.longestsubstringwithoutrepeatingcharacters;

import com.leetcode.questions.leetcode.editor.en.LongestSubstringWithoutRepeatingCharacters;
import com.leetcode.questions.support.GeneratedLeetCodeTestHarness;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.nio.file.Files;
import java.nio.file.Path;

public class LongestSubstringWithoutRepeatingCharactersTest {

    private static final Path CASES_PATH = Path.of("generated-test-cases", "longest-substring-without-repeating-characters.json");

    public LongestSubstringWithoutRepeatingCharactersTest() {

    }

    @Test
    public void testCases() throws Exception {
        GeneratedLeetCodeTestHarness.run(LongestSubstringWithoutRepeatingCharacters.Solution.class, "lengthOfLongestSubstring", Files.readString(CASES_PATH));
    }
}

File: leetcode-questions/csharp/src/test/csharp/com/leetcode/questions/longestsubstringwithoutrepeatingcharacters/LongestSubstringWithoutRepeatingCharactersTest.cs
using Xunit;
using System.Collections.Generic;
using com.leetcode.questions.leetcode.editor.en.longestsubstringwithoutrepeatingcharacters;
using com.leetcode.questions.support;

namespace com.leetcode.questions.longestsubstringwithoutrepeatingcharacters;

public class LongestSubstringWithoutRepeatingCharactersTest {

    private const string CasesFileName = "longest-substring-without-repeating-characters.json";

    public LongestSubstringWithoutRepeatingCharactersTest() {

    }

    [Fact]
    public void TestCases() {
        GeneratedLeetCodeTestHarness.Run(typeof(Solution), "LengthOfLongestSubstring", GeneratedLeetCodeTestHarness.ReadCasesFile(CasesFileName));
    }
}
